Samples/WMCConsole_winrtcpp/WMCConsole_winrtcpp.cpp

```diff
@@ -0,0 +1,255 @@
+//
+#include <Windows.h>
+#include <iostream>
+#include <conio.h>
+#include <Mferror.h>
+#include <winrt\base.h>
+#include <winrt\Windows.Foundation.h>
+#include <winrt\Windows.Foundation.Collections.h>
+#include <winrt\Windows.Media.Capture.h>
+#include <winrt\Windows.Media.Core.h>
+#include <winrt\Windows.Media.Devices.h>
+#include <winrt\Windows.Media.MediaProperties.h>
+#include <winrt\Windows.Media.Capture.Frames.h>
+
+#include <winrt\Windows.Devices.Enumeration.h>
+#include <winrt\Windows.Graphics.Imaging.h>
+#include <winrt\Windows.Storage.h>
+#include <winrt\Windows.Storage.Streams.h>
+
+using namespace winrt;
+using namespace Windows::Media::Capture;
+using namespace Windows::Media::MediaProperties;
+using namespace Windows::Media::Capture::Frames;
+using namespace Windows::Foundation;
+using namespace Windows::Foundation::Collections;
+using namespace Windows::Devices::Enumeration;
+using namespace Windows::Storage;
+using namespace Windows::Graphics::Imaging;
+using namespace Windows::Media::Core;
+IVectorView<MediaFrameSourceGroup> sourceGroups;
+// Iterates through source groups and filter-out the frame sources which have IR,depth and other sources which we cannot consume in this app
+IVector<MediaFrameSourceInfo> GetFilteredSourceGroupList()
+{
+    auto filteredSourceInfos = single_threaded_vector<MediaFrameSourceInfo>();
+    sourceGroups = MediaFrameSourceGroup::FindAllAsync().get();
+    auto sourceGroupIter = sourceGroups.First();
+    while (sourceGroupIter.HasCurrent())
+    {
+        std::wcout << sourceGroupIter.Current().DisplayName().c_str();
+        auto sourceInfos = sourceGroupIter.Current().SourceInfos();
+        auto sourceInfoIter = sourceInfos.First();
+        // iterate through sources and filter-out the IR,depth and other sources which we cannot consume in this app
+        while (sourceInfoIter.HasCurrent())
+        {
+            if ((sourceInfoIter.Current().MediaStreamType() == MediaStreamType::Photo
+                || sourceInfoIter.Current().MediaStreamType() == MediaStreamType::VideoPreview
+                || sourceInfoIter.Current().MediaStreamType() == MediaStreamType::VideoRecord)
+                && sourceInfoIter.Current().SourceKind() == MediaFrameSourceKind::Color)
+            {
+                filteredSourceInfos.Append(sourceInfoIter.Current());
+            }
+            sourceInfoIter.MoveNext();
+        }
+        sourceGroupIter.MoveNext();
+    }
+    return filteredSourceInfos;
+}
+
+int GetSGSelection(IVector<MediaFrameSourceInfo> filteredGroup)
+{
+    unsigned int idx = 0;
+    auto group = filteredGroup.First();
+    while (group.HasCurrent())
+    {
+        idx++;
+        auto currGroup = group.Current();
+        if (currGroup.SourceGroup() == nullptr)
+        {
+            std::cout << "nullptr";
+        }
+        // These are in the same order as the enum MediaStreamType
+        // TODO: better solution is to create a Pair type lookup which is future proof

```

Comment:
>// TODO: better solution is to create a Pair type lookup which is future proof [](start = 8, length = 78)

remove / address?

Reply:
same goes for panelTypes, I agree that a map<int, string> would be better

---
In reply to: [284523003](https://github.com/microsoft/Windows-Camera/pull/9#discussion_r284523003) [](ancestors = 284523003)